src/main/java/org/opensearch/neuralsearch/processor/chunker/Chunker.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import java.util.Map;
+import java.util.List;
+
+/**
+ * The interface for all chunking algorithms.
+ * All algorithms need to validate parameters and chunk the content.
+ */
+public interface Chunker {
+
+    /**
+     * Validate the parameters for chunking algorithm,
+     * will throw IllegalArgumentException when parameters are invalid
+     *
+     * @param parameters a map containing non-runtime parameters for chunking algorithms
+     */
+    void validateParameters(Map<String, Object> parameters);
+
+    /**
+     * Parse the parameters for chunking algorithm.
+     * The parameters must be validated before parsing.
+     *
+     * @param parameters a map containing non-runtime parameters for chunking algorithms
+     */
+    void parseParameters(Map<String, Object> parameters);
```

Comment:
Can we merge the validateParameters and parseParameters into one method? Or this is for different case? Since parsing input twice doesn't make sense to me.

Reply:
According to @martin-gaievski, it is not a good practice to mix parameter validation and parsing within the same function. That's why I separate the original `validateAndParseParameters` method into two methods.

Reply:
that's valid point, my confusion came from the initial method's name, I think this function is more like a getter, that can validate, fail if needed, or return proper value otherwise. if we rename that to something like getParameter<...> I'm fine with having everything in on method

Reply:
I have removed the method for validate parameter. The util class for parameter validation has been merged to parameter parser.